src/main.c

```diff
@@ -347,6 +385,66 @@ static void callback(const nrsc5_event_t *evt, void *opaque)
     }
 }
 
+#ifndef __MINGW32__
+static void restore_termios(void *arg)
+{
+    tcsetattr(STDIN_FILENO, TCSANOW, arg);
+}
+#endif
+
+static void *input_main(void *arg)
```

Comment:
Make `arg` a `state_t *`?

Reply:
Never mind, it's `void *` because it's invoked by pthreads.

Reply:
Yeah, I considered assigning it to a local variable just to make it explicit that it should be a ```state_t *``` rather than rely on the implicit conversion.